gon/compound.py

```diff
@@ -0,0 +1,74 @@
+from abc import abstractmethod
+
+from orient.planar import Relation
+
+from .geometry import Geometry
+from .hints import Coordinate
+
+Relation = Relation
+
+
+class Compound(Geometry):
+    @abstractmethod
+    def __contains__(self, other: 'Geometry') -> bool:
+        """
+        Checks if the geometry contains the other.
+        """
+
+    @abstractmethod
+    def __ge__(self, other: 'Compound') -> bool:
+        """
+        Checks if the geometry is a superset of the other.
+        """
+
+    @abstractmethod
+    def __gt__(self, other: 'Compound') -> bool:
+        """
+        Checks if the geometry is a strict superset of the other.
+        """
+
+    @abstractmethod
+    def __le__(self, other: 'Compound') -> bool:
+        """
+        Checks if the geometry is a subset of the other.
+        """
+
+    @abstractmethod
+    def __lt__(self, other: 'Compound') -> bool:
+        """
+        Checks if the geometry is a strict subset of the other.
+        """
+
+    def disjoint(self, other: 'Compound') -> bool:
+        return self.relate(other) is Relation.DISJOINT
+
+    @abstractmethod
+    def relate(self, other: 'Compound') -> Relation:
+        """
+        Finds relation between geometric objects.
+        """
+
+
+class Linear(Geometry):
+    @property
+    @abstractmethod
+    def length(self) -> Coordinate:
+        """
+        Returns length of the geometry.
+        """
```

Comment:
I don't think linear objects should have `area` property, it doesn't make sense for them, as `length` field doesn't make sense for `Polygon`s, only `perimeter`, but it is not a `length` of `Polygon`

I was thinking about a way to determine whether the object is `Linear` / `Shaped` / just a `Point` based on dimension instead of `isinstance` checks, something like
```python
area = intersection.area if intersection.dimension == 2 else 0
```
if it will make easier to read the code.

But then what should be a dimension of geometry collection which can be a result of polygons intersection? It's too hard for me for now.

Reply:
Interestingly enough, both area and length are defined for all geometry types in Shapely: https://shapely.readthedocs.io/en/latest/manual.html#general-attributes-and-methods

Reply:
well, that's odd, we definitely do not want to pollute classes namespaces with things that make no sense to them